tests/test_cnes_sonar_scanner.py

```diff
@@ -0,0 +1,517 @@
+        """
+        This function waits for SonarQube to be configured by
+        the configure.bash script.
+
+        :param container_name: name of the container running lequal/sonarqube
+        :param tail: forwarded to docker logs
+        """
+        docker_client = docker.from_env()
+        while b'[INFO] CNES SonarQube: ready!' not in docker_client.containers.get(container_name).logs(tail=tail):
+            time.sleep(10)
+    
+    @classmethod
+    def language(cls, language_name: str, language_key: str, folder: str,
+        sensors_info, project_key: str, nb_issues: int, cnes_qp: str = "",
+        nb_issues_cnes_qp: int = 0):
+        """
+        This function tests that the image can analyze a project.
+
+        Environment variables used:
+            SONARQUBE_URL
+            SONARQUBE_LOCAL_URL
+            SONARQUBE_NETWORK
+            SONARQUBE_ADMIN_PASSWORD
+
+        :param language_name: language name to display
+        :param language_key: language key for SonarQube
+        :param folder: folder name, relative to the tests/ folder
+        :param sensors_info: array of lines of sensors to look for in the scanner output
+        :param project_key: project key (sonar.project_key of sonar-project.properties)
+        :param nb_issues: number of issues with the Sonar way Quality Profile
+        :param cnes_qp: (optional) name of the CNES Quality Profile to apply, if any
+        :param nb_issues_cnes_qp: (optional) number of issues with the CNES Quality Profile, if specified
+
+        Example (not a doctest):
+            sensors = (
+                "INFO: Sensor CheckstyleSensor [checkstyle]",
+                "INFO: Sensor FindBugs Sensor [findbugs]",
+                "INFO: Sensor PmdSensor [pmd]",
+                "INFO: Sensor CoberturaSensor [cobertura]"
+            )
+            self.language("Java", "java", "java", sensors, "java-dummy-project", 3, "CNES_JAVA_A", 6)
+        """
+        docker_client = docker.from_env()
+        print(f"Analysing project {project_key}...")
+        output = docker_client.containers.run(cls._SONAR_SCANNER_IMAGE, f"-Dsonar.projectBaseDir=/usr/src/tests/{folder}",
+            auto_remove=True,
+            environment={"SONAR_HOST_URL": cls.SONARQUBE_URL},
+            network=cls.SONARQUBE_NETWORK,
+            user=f"{os.getuid()}:{os.getgid()}",
+            volumes={
+                f"{cls._PROJECT_ROOT_DIR}": {'bind': '/usr/src', 'mode': 'rw'},
+                f"{cls._PROJECT_ROOT_DIR}/.sonarcache": {'bind': '/opt/sonar-scanner/.sonar/cache', 'mode': 'rw'}
+            }).decode("utf-8")
+        print(output)
+        # Make sure all non-default for this language plugins were executed by the scanner
+        for sensor_line in sensors_info:
+            # Hint: if this test fails, a plugin may not be installed correctly or a sensor is not triggered when needed
+            assert sensor_line in output
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/cnescatlab/sonar-scanner/pullRequest?prid=6252273)